### DISCUSSION TITLE
Trip Recording, Webhook, Online Tracking, format and units of data

### DISCUSSION THREAD

**Discussion by Type-Style** [osmandapp/OsmAnd#17607] [Q&A] [answered]
Hi there Osmand Community,

I am using the Osmand+ App on Android, along with the "Triprecording"
Plugin.
In the Plugin, I am using the online tracking. Which is basicly a
webhook.

I am able to recieve the data, so no issue here.
But I am unsure about the format or units of the data I am receving.
The documentation is not quite clear in this section:
https://www.osmand.net/docs/user/plugins/trip-recording/

> The Web address field is for entering the URL using the parameter
> syntax: lat={0}, lon={1}, timestamp={2}, hdop={3}, altitude={4}, speed={5}, bearing={6}

Lets quickly go through the parameters to show where the confusion is
coming from:
**Lat** (is fine)
**Lon** (is clear too)
**timestamp**: _is confusing to me, it looks like a unix timestamp but _
can't quite tell. What format is that?
Here is an example: 1688308684225 (received on the 02.07) if
interpreting as unix time it shows a data far in the future. Could you
clarify about the format?_
**hdop** (clear)
**altitude** (expecations: it is meters above see level?), Why do I get a
minimum of 40m then even on the sea itself? Maybe it has something to do
with how it is calculating the height above ground? I am somewhat aware
of the different theoretical geoid models. Could you point me in the
right direction of what is used and how it is calculated?
**speed** (is is m/s?)
**bearing**: is it the bearing as seen in on the display widget? Meaning a
relative direction towards a target? Or is it comapred to north in other
words a "heading"?

Thanks for taking the time.
I like the app a lot!

Kind Regards

**Comment by vshcherb** [comment, marked as answer]
speed - m/s, altitude - same as displayed in android widget (so it depends whether correction is applied or not), bearing is actually provided by Android API (https://developer.android.com/reference/android/location/Location#getBearing()) - movement direction, timestamp - is in ms (long - Java standard System.currentTimeMillis()) - here is tested https://www.unixtimestamp.com/

**Reply by Type-Style** [reply]
Thanks this helps.
What about the height if I may ask. Any hints or direction in which to research?

**Reply by vshcherb** [reply]
altitude - same as displayed in android widget (so it depends whether correction is applied or not) - it should be documented somehow because it's tricky algorithm
